### DISCUSSION TITLE
Three necessities to switch back from VS Code to Atom

### DISCUSSION THREAD

**Discussion by trusktr** [atom/atom#23171] [Ideas]
1. Speed, at least as fast as VS Code. I can't live without the speed of VS Code.
2. Highly-integrated built-in terminal, like VS Code, so that editor extensions have a standard terminal to build on top of rather than relying on specific 3rd-party terminals. Because VS Code has a built-in terminal, anything that plugins and extensions can do with it via extension API is standard and guaranteed to be supported and won't break, and any features that VS Code builds (f.e. multiplayer code editing with buddies) works and the terminal is part of the experience.
3. TypeScript support needs to be on par with VS Code, and preferably built-in.

If atom can have those three things, I would switch back for sure.

I really love Atom's UX and visual design. It is so much nicer than VS Code.

Also Atom's vim-mode-plus extension **is the best version of Vim that exists on Earth**, so good that I would rather script vim-mode-plus than actual Vim (it can do everything Vim/Neovim can do, and more, and with better modern graphical effects).

However I can't bare the slowness of Atom (last time I tried), and I dislike the lack of highly-integrated features around the terminal, etc.

VS Code functionality out of the box makes me more productive, without having to install 3rd-party plugins that don't work well with each other.

Take this lightly, it's just an openly honest description.

**Comment by ThatXliner** [comment]
> without having to install 3rd-party plugins that don't work well with each other.

Try out https://atom-community.io

> However I can't bare the slowness of Atom (last time I tried)

Atom is designed to be very customizable. This also means the plugins can modify the UI, [unlike VSCode](https://code.visualstudio.com/api/advanced-topics/extension-host#stability-and-performance), which can reduce performance. For me right now, Atom's fast enough

> Highly-integrated built-in terminal

https://github.com/bus-stop/terminus is one extension I use. It's forked from https://github.com/platformio/platformio-atom-ide-terminal which is forked from https://github.com/jeremyramin/terminal-plus. They expose a similar (if not the same) API which a lot of extensions support. But yeah, it's not built-in which I guess is a downside. People had ideas like this as well
 - https://github.com/atom/atom/issues/14490
